src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,41 @@
 package mate.academy;
 
+import java.util.Map;
+import java.util.Random;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Random random = new Random();
+    private Map<String, UserData> map = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        try {
+            Thread.sleep(1000);
```

Comment:
You don't need to wait each time, because the second time you receive a request with the same userId it should return response immediately. So trigger this 'work simulation' only when there is not processed request for the user yet.